library/src/main/java/com/github/gfx/android/orma/SingleAssociation.java

```diff
@@ -111,26 +122,27 @@ public long getId() {
         return id;
     }
 
-    @CheckResult
-    @NonNull
-    public Single<Model> single() {
-        return single;
-    }
-
     /**
-     * A shortcut of {@code singleAssociation.single().blockingGet()}.
+     * Get the model referred from this instance with blocking.
      *
      * @return A model that the instance refers to.
      */
     @NonNull
     public Model get() throws NoValueException {
-        return single.blockingGet();
+        if (value == null) {
+            synchronized (this) {
+                if (value == null) {
+                    value = factory.call();
+                }
+            }
+        }
+        return value;
```

Comment:
I rewrite lazy and cached `get()` methods via double-checked technique instead `Single#blockingGet()`.